go/libraries/doltcore/merge/merge_prolly_rows.go

```diff
@@ -712,6 +720,62 @@ func remapTuple(tuple val.Tuple, desc val.TupleDesc, mapping val.OrdinalMapping)
 	return result
 }
 
+// remapTupleWithColumnDefaults takes the given |tuple| (and the |tupleDesc| that describes how to access its fields)
+// and uses |mapping| to map the tuple's data and return a new tuple. |tm| provides high access to the name of the table
+// currently being merged and associated node store. |mergedSch| is the new schema of the table and is used to look up
+// column default values to apply to any existing rows when a new column is added as part of a merge. |pool| is used to
+// to allocate memory for the new tuple. A pointer to the new tuple data is returned, along with any error encountered.
+func remapTupleWithColumnDefaults(ctx context.Context, tuple *val.Tuple, tupleDesc val.TupleDesc, mapping val.OrdinalMapping, tm *TableMerger, mergedSch schema.Schema, pool pool.BuffPool) (*val.Tuple, error) {
+	tb := val.NewTupleBuilder(mergedSch.GetValueDescriptor())
+
+	for to, from := range mapping {
+		var value interface{}
+		if from == -1 {
+			// If the column is a new column, then look up any default value
+			col := mergedSch.GetNonPKCols().GetByIndex(to)
+			if col.Default != "" {
+				// We need a sql.Context to call parse.StringToColumnDefaultValue, so see if we have one already;
+				// if not, create one, since this code also gets called from the CLI merge code path.
+				sqlCtx, ok := ctx.(*sql.Context)
+				if !ok {
+					sqlCtx = sql.NewContext(ctx)
```

Comment:
creating a new sql context is surprisingly expensive due to the cost of creating the embedded sql session, we should create this once per table/root if possible.

Reply:
Oh, thanks for pointing that out! I just pushed a new commit that moves this out of the loop and reuses a single instance for the entire table merge.